cr-examples/onnx/src/main/java/oracle/code/onnx/OnnxRuntime.java

```diff
@@ -55,58 +55,79 @@ public final class OnnxRuntime {
     public interface OnnxFunction<T> extends Supplier<T>, Quotable {
     }
 
+    record CachedModel(byte[] protoModel, int[] operandsMapping) {}
+
+    private static final WeakHashMap<Class<?>, CachedModel> MODEL_CACHE = WeakHashMap.newWeakHashMap(10);
```

Comment:
Using `ClassValue` is another option.

Reply:
Unfortunately `ClassValue` does not support computation if absent (if more than just the bare class is needed).

Reply:
I believe `ClassValue` is similar to a thread local in how it's used -- e.g. you create an instance of it and stick it in a static final (the javadoc of `ClassValue` is useless in that respect):

```
ClassValue<CachedModel> CACHED_MODEL = new ClassValue<>() {
    @Override
    CachedModel computeValue() { <logic here> }
};
```

That should work here.

Reply:
Yes, I'm just having troubles to initiate the ClassValue when I need it (when I have additional arguments).
See my solution.

Reply:
There is a design flaw in the ClassValue, the only way to initialize it is by ClassValue::computeValue(Class<?>).
It has been designed to attach some value to a class, however if your value needs some arguments to be computed, you need to attach them to the class, which you cannot. So it is a Catch-22.

Reply:
Ah I see. I wouldn't say that's an issue with `ClassValue` -- but more of an issue with the fact that the class is not a valid key in your case (and `ClassValue` is better alternative to a  `Map<Class, V>`).

But here I wonder -- if the passed quotable instance does have an associated `Quoted` object -- doesn't it mean that it must be a class generated by the metafactory, and one that is specialized for that particular code model? E.g. I'm wondering if the class alone (after you verified there's a `Quoted` object in there) isn't already a good key?

Reply:
I tried it and unfortunately a different instances of Quoted each time.

Reply:
BTW: ThreadLocal does not have this problem, it can be set at any time by any value.

Reply:
Ok - the issue here is that you need the lookup to create the CachedModel instance